tools/base/fpgainfo-c/tempinfo.c

```diff
@@ -23,35 +23,50 @@
 // CONTRACT,  STRICT LIABILITY,  OR TORT  (INCLUDING NEGLIGENCE  OR OTHERWISE)
 // ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE,  EVEN IF ADVISED OF THE
 // POSSIBILITY OF SUCH DAMAGE.
+
+#include <getopt.h>
 #include "fpgainfo.h"
+#include "sysinfo.h"
 #include "tempinfo.h"
 #include <opae/fpga.h>
 //#include <uuid/uuid.h>
+#include <wchar.h>
 
 #include "bmcinfo.h"
-#include "sysinfo.h"
+#include "safe_string/safe_string.h"
+
+/*
+ * Print help
+ */
+void temp_help(void)
+{
+	printf("\nPrint thermal metrics\n"
+	       "        fpgainfo temp [-h]\n"
+	       "                -h,--help           Print this help\n"
+	       "\n");
+}
 
-static void print_temp_info(fpga_properties props, struct dev_list *dev)
+static void print_temp_info(fpga_properties props)
 {
 	fpga_result res = FPGA_OK;
-	char path[SYSFS_PATH_MAX];
-	uint32_t temperature = -1;
+	const char *sysfspath = get_sysfs_path(props, FPGA_DEVICE, NULL);
 
 	fpgainfo_print_common("//****** TEMP ******//", props);
 
-	if (NULL == dev) {
-		printf("  NO THERMAL DATA AVAILABLE\n");
-		return;
-	}
-
-	snprintf_s_ss(path, sizeof(path), "%s/%s", dev->sysfspath,
-		      "thermal_mgmt/temperature");
-	res = sysfs_read_u32(path, &temperature);
+	// BMC thermal sensors are more accurate than sysfs - don't print sysfs
+#ifdef PRINT_SYSFS_TEMPERATURE
+	char path[SYSFS_PATH_MAX];
+	uint32_t temperature = -1;
+	snprintf_s_ss(path, sizeof(path), "%s/%s", sysfspath,
+		      SYSFS_THERMAL_FILE);
+	res = fpgainfo_sysfs_read_u32(path, &temperature);
```

Comment:
Is there anyway that temperature can have another API call for temperature? Seems like it would make sense imo
@r-rojo

Reply:
This was in response to an email - apparently the BMC sensor has less variability than the board's sensor.  PM didn't want to display the value for the less accurate sensor.

I think OPAE is working on a metrics API that would potentially replace all of this sysfs stuff, if that's what you mean.